### ISSUE TITLE
[Bug] WAM  can hide the windows (account picker and plugin browser) behind the app

### ISSUE THREAD

**Issue by maliksahil** [AzureAD/microsoft-authentication-library-for-dotnet#2901]
**Logs and network traces**
Without logs or traces, it is unlikely that the team can investigate your issue. Capturing logs and network traces is described in [Logging wiki](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/logging).

**Which version of MSAL.NET are you using?**

4.36.1

**Platform**

.NET Core

**What authentication flow has the issue?**
* Desktop / Mobile
    * [ X] Interactive
    * [ ] Integrated Windows Authentication
    * [ ] Username Password
    * [ ] Device code flow (browserless)
* Web app 
    * [ ] Authorization code
    * [ ] On-Behalf-Of
* Daemon app 
    * [ ] Service to Service calls

Other?

WAM

**Is this a new or existing app?**

Existing and new

**Repro**

If you integrate WAM with any console app using .WithBroker or .WithWindowsBroker, the first time the wam login dialog appears, it appears below the console window - and it is not even in ALT_TAB. This gives the false impression to the user that the console app is hung, and no obvious way of finding the WAM dialog box. This has been raised as an issue by GCM, will most likely be an issue with PowerShell, and a few other customers doing the adal->msal migration.

```csharp
var your = (code) => here;
```

**Expected behavior**
A clear and concise description of what you expected to happen (or code).

The WAM dialog should either be a top level window or the console window should be it's parent.

**Actual behavior**
A clear and concise description of what happens, e.g. exception is thrown, UI freezes.  

The WAM dialog shows up randomly frequently behind the console window, and cannot be found via alt_tab and gives the user an impression that the app is frozen.

**Possible solution**

**Additional context / logs / screenshots**
Add any other context about the problem here, such as logs and screenshots.

**Comment by bgavrilMS** [comment]
Do you have a repro for this Sahil? I cannot repro this. 
In particular for GCM, they should be using `WithParentWindowOrActivity(IntPtr windowHandle)` and pass in 
the value of 

```csharp
[DllImport("kernel32.dll")]
static extern IntPtr GetConsoleWindow();
```

If `WithParentWindowOrActivity` is not passed in, then MSAL will attempt to use the "desktop" as the parent window. 

In either case, the Account Picker should be at the top.

**Comment by bgavrilMS** [comment]
I logged https://github.com/microsoft/Git-Credential-Manager-Core/issues/469 on GCM to fix this.

**Comment by bgavrilMS** [comment]
We could throw an exception if applications do not give us their window handle, but that would be a somewhat of a breaking change.

**Comment by bgavrilMS** [comment]
As discussed with Matthew, console applications cannot use `GetConsoleWidnow` to get a window handle, because this has been deprecated (see https://docs.microsoft.com/en-us/windows/console/getconsolewindow). 

`GetConsoleWindow` continues to work when using the classical "cmd" (on Win10) but does not work with the new terminal window (default on win11). We need guidance from the Windows team about this scenario. 

A workaround is as follows, which needs to be implented:

1. WinForm and WPF apps are still asked to provide a parent window
2. Console apps do not need to provide one. In which case, MSAL should: 
-  Walk up the process tree and find the first process that has a window handle (I believe the is no .NET API for this, but there are window APIs)

We still need help from the Windows team to improve the experience: 

- the Account picker should be the topmost window
- the Account picker should be alt-tababble

**Comment by bgavrilMS** [comment]
No more reports of this problem after the partial fix, closing as resolved in MSAL 4.37.
